browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -222,9 +222,14 @@ void IPFSTabHelper::UpdateDnsLinkButtonState() {
     }
     return;
   }
-
-  GURL current = web_contents()->GetURL();
-  if (ipfs_resolved_url_.is_valid() && resolver_->host() != current.host()) {
+  if (!ipfs_resolved_url_.is_valid())
+    return;
+  GURL current = GetCurrentPageURL();
+  if (resolver_->host() != current.host()) {
+    ipfs_resolved_url_ = GURL();
+    UpdateLocationBar();
+  }
+  if (!CanResolveURL(current)) {
```

Comment:
As I mentioned in DM, I would prefer combine these two cases to avoid duplicating below two lines,
```
ipfs_resolved_url_ = GURL();    
UpdateLocationBar();
```